src/dimension/mod.rs

```diff
@@ -329,5 +529,183 @@ mod test {
+                    (min2, max2, if step2 == 0 { 1 } else { step2 })
+                ) == intersects
+            )
+        }
+    }
+
+    #[test]
+    fn slice_min_max_empty() {
+        assert_eq!(slice_min_max(0, Slice::new(0, None, 3)), None);
+        assert_eq!(slice_min_max(10, Slice::new(1, Some(1), 3)), None);
+        assert_eq!(slice_min_max(10, Slice::new(-1, Some(-1), 3)), None);
+        assert_eq!(slice_min_max(10, Slice::new(1, Some(1), -3)), None);
+        assert_eq!(slice_min_max(10, Slice::new(-1, Some(-1), -3)), None);
+    }
+
+    #[test]
+    fn slice_min_max_pos_step() {
+        assert_eq!(slice_min_max(10, Slice::new(1, Some(8), 3)), Some((1, 7)));
+        assert_eq!(slice_min_max(10, Slice::new(1, Some(9), 3)), Some((1, 7)));
+        assert_eq!(slice_min_max(10, Slice::new(-9, Some(8), 3)), Some((1, 7)));
+        assert_eq!(slice_min_max(10, Slice::new(-9, Some(9), 3)), Some((1, 7)));
+        assert_eq!(slice_min_max(10, Slice::new(1, Some(-2), 3)), Some((1, 7)));
+        assert_eq!(slice_min_max(10, Slice::new(1, Some(-1), 3)), Some((1, 7)));
+        assert_eq!(slice_min_max(10, Slice::new(-9, Some(-2), 3)), Some((1, 7)));
+        assert_eq!(slice_min_max(10, Slice::new(-9, Some(-1), 3)), Some((1, 7)));
+        assert_eq!(slice_min_max(10, Slice::new(1, None, 3)), Some((1, 7)));
+        assert_eq!(slice_min_max(10, Slice::new(-9, None, 3)), Some((1, 7)));
+        assert_eq!(slice_min_max(11, Slice::new(1, None, 3)), Some((1, 10)));
+        assert_eq!(slice_min_max(11, Slice::new(-10, None, 3)), Some((1, 10)));
+    }
+
+    #[test]
+    fn slice_min_max_neg_step() {
+        assert_eq!(slice_min_max(10, Slice::new(1, Some(8), -3)), Some((1, 7)));
+        assert_eq!(slice_min_max(10, Slice::new(2, Some(8), -3)), Some((4, 7)));
+        assert_eq!(slice_min_max(10, Slice::new(-9, Some(8), -3)), Some((1, 7)));
+        assert_eq!(slice_min_max(10, Slice::new(-8, Some(8), -3)), Some((4, 7)));
+        assert_eq!(slice_min_max(10, Slice::new(1, Some(-2), -3)), Some((1, 7)));
+        assert_eq!(slice_min_max(10, Slice::new(2, Some(-2), -3)), Some((4, 7)));
+        assert_eq!(slice_min_max(10, Slice::new(-9, Some(-2), -3)), Some((1, 7)));
+        assert_eq!(slice_min_max(10, Slice::new(-8, Some(-2), -3)), Some((4, 7)));
+        assert_eq!(slice_min_max(9, Slice::new(2, None, -3)), Some((2, 8)));
+        assert_eq!(slice_min_max(9, Slice::new(-7, None, -3)), Some((2, 8)));
+        assert_eq!(slice_min_max(9, Slice::new(3, None, -3)), Some((5, 8)));
+        assert_eq!(slice_min_max(9, Slice::new(-6, None, -3)), Some((5, 8)));
+    }
+
+    #[test]
+    fn slices_intersect_true() {
+        assert!(slices_intersect(
+            &Dim([4, 5]),
+            &[(..).into(), (..).into()],
+            &[(..).into(), (..).into()],
+        ));
+        assert!(slices_intersect(
+            &Dim([4, 5]),
+            &[0.into(), (..).into()],
+            &[0.into(), (..).into()],
```

Comment:
This is not a very interesting comment, but I will note my passing disapproval of Rust code that ends up looking like x.into(), y.into() where it is not about argument conversion (inside the function receiving the parameters).

This is testing code, and style doesn't matter so much here, so it's not about that. We need to design an interface that doesn't encourage the .into() style. multislice doesn't, so that's good.

The disapproval is founded on a) easier to read the X::from(..) variant and b) Seeing breaking changes due to once-unambiguous `.into()` calls fail with minor changes in Rust or other crates.

Reply:
This is something I should keep in mind more. The biggest thing I dislike about `.into()` is that you have to look at the docs to figure out what type it's converting into, but that's also a good point about issues with type inference due to minor changes.

In this case, it's actually possible to use the `s![]` macro, which is something I didn't consider when writing the PR. I think the `s![]` macro will work for most cases in similar client code. I just pushed another commit to this PR switching to `s![]`.